### PR TITLE
Fix regression with setting empty environment variable

### PR DESCRIPTION
**Pull request by howardjohn**
This was broken with ProxyConfig supported added

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -170,17 +170,18 @@ pub fn construct_config(pc: ProxyConfig) -> Result<Config, Error> {
     } else {
         "https://localhost:15012".to_string()
     };
-
-    let xds_address = parse(XDS_ADDRESS)?
-        .or(pc.discovery_address)
-        .or_else(|| Some(default_istiod_address.clone()));
+    let xds_address = empty_to_none(
+        parse(XDS_ADDRESS)?
```
Comment (hzxuzhonghu) [inline src/config.rs:174]: why not fix the parse and parse default?

Comment (howardjohn) [reply src/config.rs:174]: parse distinguishes between unset (None) and set to "" (Some("")). I don't think that's a bug, just different use case

Comment (hzxuzhonghu) [reply src/config.rs:174]: But we do not need to distinguish unset and empty "".

Comment (howardjohn) [reply src/config.rs:174]: XDS is disabled by setting to "". Unset means to use the default address

File: src/config.rs [hunk 2]
```diff
@@ -314,6 +315,15 @@ fn construct_proxy_config(mc_path: &str, pc_env: Option<&str>) -> anyhow::Result
     Ok(pc)
 }
 
+pub fn empty_to_none<A: AsRef<str>>(inp: Option<A>) -> Option<A> {
```
Comment (hzxuzhonghu) [inline src/config.rs:318]: nit since it is Option<string>, can rewrite as 
pub fn empty_to_none (inp: Option<String>) -> Option<String> {

**Review by stevenctl — APPROVED** [review]

**Pull request merged**